### ISSUE TITLE
[Feature]Mission attribute to suppress map flags

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#6760]
### Problem Description

I'm frustrated when I try to make a mission sending the player to search for something or go find a planet that matches certain characteristics, and there is no way to simply make the game *not* display the flags. 

Currently, the only way to do so is to make the mission invisible; but that also hides the mission from the mission list, which is undesireable.

### Related Issue Links

Can't find any right now, although I'm sure I've seen this come up in the past.

### Desired Solution

My request is to have six additional mission attributes added:
`"no destination flag"`
`"no stopover flag"`
`"no waypoint flag"`
`"visible destination flag" <system>`
`"visible stopover flag" <system>`
`"visible waypoint flag" <system>`

Each of these attributes causes that mission to not display the map flag for that particular element of the mission, and the second set would cause the map to display a flag on the specified system *instead* of on the one the mission would normally display.

### Alternative Approaches

Having an "alternative flag" that can accept a list of systems to display generic waypoint/destination style flags could simpify things a bit, but ideally I would want the destination flag to follow the same behavior as the regular one, being dimmer until the conditions are met, and brighter when the player can just go straight there and be done.

### Additional Context

In the particular case of the mission I am working on, the Remnant are sending the player to scan a ship and return to the starting planet. They don't know exactly where the ship is, though, so the player will have to search through a few systems. So I'd like to be able to put a flag up in a system in the center of the search area. (the ship in question is hardcoded to a specific system for story reasons, but in-game neither the Remnant nor the player have any way of knowing that's where it will be. As far as both groups are concerned, this is just a go out, scan, return to base thing.).

However, once they have scanned the ship, they get a scan message telling them to land immediately, and the mission ends there, with the next mission triggering on landing for the next part of the story.

**Comment by tehhowch** [comment]
1st thought re: implementation, if you can't use a hidden mission + active mission, is that these should not be Mission fields. These are highly related fields that belong on some struct e.g. `guidance`, and missions have one of this with appropriate defaults.

**Comment by Amazinite** [comment]
> Currently, the only way to do so is to make the mission invisible; but that also hides the mission from the mission list, which is undesireable.

You can also create two missions. The first mission tells you the objective and never completes and the second mission has the true destination. The first mission can then be failed when the second mission is completed.

**Comment by Zitchas** [comment]
Yes, this is what I currently do. It seems rather clumsy to have to use two missions to do what really should be a single "don't show a flag for this" option.

**Comment by Amazinite** [comment]
The new `complete at` node can be used to hide the destination of a mission, as `complete at` nodes do not create map markers while superseding any `destination` node.
`on enter` actions can be used to set conditions that are required to complete a mission, effectively acting as hidden `waypoint`s.
The `on land` action requested in #4978 could be used to hide stopovers in the same way that `on enter` hides waypoints. But just as I described that separate missions can be used to "hide" destinations, they can also be used to "hide" stopovers right now.
